badges/tests/test_commands.py

```
@@ -0,0 +1,501 @@
+def test_reconcile_accepts_a_member_by_primary_key(
+    plain_user, commit_by_someone_else, stale_commit_grant
+):
+    """An id is as good as an email, because an admin URL only carries the id."""
+    call_command("reconcile_achievements", "--user", str(plain_user.pk))
+
+    assert not UserAchievement.objects.filter(user=plain_user).exists()
+
+
+def test_reconcile_rejects_an_unknown_member(plain_user):
+    """A typo in ``--user`` must not silently widen the run to everybody."""
+    with pytest.raises(CommandError, match="nobody@example.com"):
```

Ruff (0.16.0)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)